### PR TITLE
Update lib versions to support python 3.11+

### PR DESCRIPTION
**Pull request by Alexander-Belyi**
We need to update pybind11 to support Python versions 3.11 and above.
Also, updating some of the other libraries to recent versions.
An update in NetworkX changed the karate club graph to its weighted version, so our tests had to be updated.
I'm also dropping support of Python 3.7 since many other libraries stopped supporting it too.

### CODE REVIEW

File: pycombo/misc.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def deconstruct_graph(graph, weight: Optional[str] = None) -> Tuple[nodes_, edge
         if is_weighted(graph, weight=weight):
             default_ = 0
         else:
-            logger.info(f"No property found: `{weight}`. Using as unweighted graph")
```
Comment (Casyfill) [inline pycombo/misc.py:83]: Why did you make than change? Personally I'd prefer backtick as in markdown it highlights code/numeric values

**Comment by Casyfill** [comment]
Thanks Alexander! All looks good, will take another look tomorrow and should be good to merge

**Review by Casyfill — APPROVED** [review]
LGTM, feel free to merge.

By the way, do you have any suggestion - I'd love to add more networks for benchmarking

File: pycombo/misc.py (hunk 1 above)
Comment (Alexander-Belyi) [reply pycombo/misc.py]: Flake8 4.0.1 on Python 3.12 complained about these backticks, so I changed them. Apparently, that was a bug, which was fixed in newer versions of flake8. I reverted this change.

**Comment by Alexander-Belyi** [comment]
Updated release actions in the same PR so we can test the automatic run of test release on the PR merge.

**Comment by Alexander-Belyi** [comment]
> By the way, do you have any suggestion - I'd love to add more networks for benchmarking

That'd be great! We actually did something similar a couple of years ago for another paper with Stan. I believe that it should be pretty straightforward to migrate the evaluation code from https://github.com/Alexander-Belyi/GNNS/blob/master/GNNS.ipynb here. I'm not sure how much time I can dedicate to helping with that, but please feel free to start the work, and I'd be happy to help (at least by reviewing =).

**Pull request merged**